Explorer/Assets/DCL/AvatarRendering/Emotes/Systems/CharacterEmoteSystem.cs

```diff
@@ -51,12 +55,34 @@ protected override void Update(float t)
         {
             ConsumeEmoteIntentQuery(World);
             ReplicateLoopingEmotesQuery(World);
+            CancelEmotesByDeletionQuery(World);
+            CancelEmotesByTeleportIntentionQuery(World);
             CancelEmotesByMovementQuery(World);
             CancelEmotesQuery(World);
             UpdateEmoteTagsQuery(World);
             CleanUpQuery(World);
         }
 
+        [Query]
+        [All(typeof(DeleteEntityIntention))]
+        private void CancelEmotesByDeletion(ref CharacterEmoteComponent emoteComponent, in IAvatarView avatarView)
+        {
+            EmoteReferences? emoteReference = emoteComponent.CurrentEmoteReference;
+
+            if (emoteReference != null)
+                StopEmote(ref emoteComponent, emoteReference, avatarView);
+        }
+
+        [Query]
+        [All(typeof(PlayerTeleportIntent))]
+        private void CancelEmotesByTeleportIntention(ref CharacterEmoteComponent emoteComponent, in IAvatarView avatarView)
+        {
+            EmoteReferences? emoteReference = emoteComponent.CurrentEmoteReference;
+
+            if (emoteReference != null)
+                StopEmote(ref emoteComponent, emoteReference, avatarView);
```

Comment:
what do you think of inserting this check inside StopEmote method? Then you can simplify both calls just to 
`StopEmote(ref emoteComponent, emoteComponent.CurrentEmoteReference, avatarView);` 
I just don't know if it correct for other use-cases of this function